xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-upload/plugin.js

```diff
@@ -162,7 +162,7 @@
         var input = $('<input>').attr({
           'type': 'hidden',
           'name': 'uploadedFiles',
-          'value': event.data.fileLoader.fileName
+          'value': JSON.parse(event.data.fileLoader.xhr.responseText).fileName
```

Comment:
The old code didn't work? Or is this bulletproofing?

Reply:
It's required, the old code was retrieving the submitted name (i.e., `__fileCreatedFromDataURI__.xxx`) instead of the attachment name returned by `$services.temporaryAttachments.uploadTemporaryAttachment`